src/Stratis.Bitcoin.Features.PoA/Voting/ConsensusRules/PoAVotingCoinbaseOutputFormatRule.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.PoA.Voting.ConsensusRules
+{
+    /// <summary>Validates <see cref="VotingData"/> collection format if voting output is present in the coinbase transaction.</summary>
+    public class PoAVotingCoinbaseOutputFormatRule : PartialValidationConsensusRule
+    {
+        private VotingDataEncoder votingDataEncoder;
+
+        public override void Initialize()
+        {
+            this.votingDataEncoder = new VotingDataEncoder(this.Parent.LoggerFactory);
```

Comment:
If the decoder needs to be used outside of the rule it will be a bit hard to call it (unless its state-less)